File: branches/BibleWebService_trunk_2.0.0.0_CSharpConversion/Chapters.vb
Imports System.Data.SqlClient
Imports System.Configuration

Public Class Chapters
    Inherits CollectionBase

    Default Public Property Item(ByVal index As Integer) As Chapter
        Get
            Return CType(list(index), Chapter)
        End Get
        Set(ByVal Value As Chapter)
            list(index) = Value
        End Set
    End Property

    Public Function Add(ByVal c As Chapter) As Integer
        Return list.Add(c)
    End Function

    Public Sub Remove(ByVal c As Chapter)
        list.Remove(c)
    End Sub

    Public Function Find(ByVal BibleID As Integer, ByVal BookID As Integer, ByVal PopulateVerses As Boolean) As Boolean
		Dim dtr As SqlDataReader = Nothing

        Try
            Dim strSQL As String = "bible_GetChapters"
			Dim cnn As New SqlConnection(ConfigurationManager.AppSettings("DataConn"))
            Dim cmd As New SqlCommand(strSQL, cnn)
            Dim blnHasRows As Boolean

            cmd.CommandType = CommandType.StoredProcedure
            cmd.Parameters.Add(New SqlParameter("@BibleID", BibleID))
            cmd.Parameters.Add(New SqlParameter("@BookID", BookID))

            cnn.Open()
            dtr = cmd.ExecuteReader(CommandBehavior.CloseConnection)

            Do While dtr.Read
                Dim c As New Chapter
                Dim vs As New Verses

                c.Chapterno = dtr.GetInt32(0)

                If PopulateVerses Then
                    vs.Find(BibleID, BookID, c.ChapterNo)
                    c.Verses = vs
                End If

                list.Add(c)
            Loop

            blnHasRows = dtr.HasRows
            dtr.Close()
            Return blnHasRows
		Catch ex As Exception
			If dtr IsNot Nothing Then
				If Not dtr.IsClosed Then
					dtr.Close()
				End If
			End If

			Dim c As New Chapter
			c.ChapterNo = -1
			List.Add(c)
			Return False
        End Try
    End Function
End Class

File: branches/BibleWebService_trunk_2.0.0.0_CSharpConversion/Chapters.cs
using System;
using System.Collections;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;

namespace BibleWebService
{
    public class Chapters : CollectionBase
    {
        public Chapter this[int index]
        {
            get { return this[index]; }
            set { this[index] = value; }
        }

        //public Chapter Item {
        //    get { return (Chapter)list(index); }
        //    set { list(index) = value; }
        //}

        public int Add(Chapter c)
        {
            return Add(c);
        }

        public void Remove(Chapter c)
        {
            Remove(c);
        }

        public bool Find(int BibleID, int BookID, bool PopulateVerses)
        {
            SqlDataReader dtr = null;

            try
            {
                string strSQL = "bible_GetChapters";
                SqlConnection cnn = new SqlConnection(ConfigurationManager.AppSettings["DataConn"]);
                SqlCommand cmd = new SqlCommand(strSQL, cnn);
                bool blnHasRows;

                cmd.CommandType = CommandType.StoredProcedure;
                cmd.Parameters.Add(new SqlParameter("@BibleID", BibleID));
                cmd.Parameters.Add(new SqlParameter("@BookID", BookID));

                cnn.Open();
                dtr = cmd.ExecuteReader(CommandBehavior.CloseConnection);

                while (dtr.Read())
                {
                    Chapter c = new Chapter();
                    Verses vs = new Verses();

                    c.ChapterNo = dtr.GetInt32(0);

                    if (PopulateVerses)
                    {
                        vs.Find(BibleID, BookID, c.ChapterNo);
                        c.Verses = vs;
                    }

                    Add(c);
                }

                blnHasRows = dtr.HasRows;
                dtr.Close();
                return blnHasRows;
            }
            catch (Exception)
            {
                if (dtr != null)
                {
                    if (!dtr.IsClosed)
                    {
                        dtr.Close();
                    }
                }

                Chapter c = new Chapter();
                c.ChapterNo = -1;
                List.Add(c);
                return false;
            }
        }
    }
}